course-v2/assets/js/video_download_popup.js

```diff
@@ -1,16 +1,22 @@
 export const initVideoDownloadPopup = () => {
-  const downloadIcon = document.querySelector(".video-download-icons")
-  const popup = document.querySelector(".video-tab-download-popup")
-  if (downloadIcon) {
-    downloadIcon.addEventListener("click", event => {
-      event.stopPropagation()
-      if (popup) popup.classList.toggle("hidden")
+  const downloadIcons = document.querySelectorAll(".video-download-icons")
+  const popups = document.querySelectorAll(".video-tab-download-popup")
+
+  if (downloadIcons.length > 0) {
+    downloadIcons.forEach((downloadIcon, index) => {
+      const popup = popups[index]
+      downloadIcon.addEventListener("click", event => {
+        event.stopPropagation()
+        if (popup) popup.classList.toggle("hidden")
+      })
     })
   }
 
   document.addEventListener("click", () => {
-    if (popup) {
-      popup.classList.add("hidden")
-    }
+    popups.forEach(popup => {
+      if (popup) {
```

Comment:
It's safe to skip the `if(popup)` here: We needed it before because `document.querySelector` can return null. But `querySelectorAll` will always return a list.